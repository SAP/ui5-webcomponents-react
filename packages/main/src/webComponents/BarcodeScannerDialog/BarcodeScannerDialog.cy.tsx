import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { BarcodeScannerDialog } from './index';

describe('BarcodeScannerDialog', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<BarcodeScannerDialog />);
    cy.get('[ui5-barcode-scanner-dialog]').should('exist');
  });

  cypressPassThroughTestsFactory(BarcodeScannerDialog);
});
