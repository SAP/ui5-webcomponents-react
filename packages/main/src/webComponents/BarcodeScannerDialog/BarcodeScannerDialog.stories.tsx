import type { Meta, StoryObj } from '@storybook/react';
import { useEffect, useRef, useState } from 'react';
import { Button } from '../Button/index.js';
import { BarcodeScannerDialog } from './index.js';

const meta = {
  title: 'Inputs / BarcodeScannerDialog',
  component: BarcodeScannerDialog,
  argTypes: {},
  args: {},
  tags: ['package:@ui5/webcomponents-fiori']
} satisfies Meta<typeof BarcodeScannerDialog>;

export default meta;
type Story = StoryObj<typeof meta>;
// TODO: check story for outdated info
export const Default: Story = {
  render(args) {
    const [open, setOpen] = useState(args.open);

    useEffect(() => {
      setOpen(args.open);
    }, [args.open]);

    return (
      <>
        <Button
          onClick={() => {
            setOpen(true);
          }}
        >
          Open BarcodeScannerDialog
        </Button>
        <BarcodeScannerDialog
          {...args}
          open={open}
          onClose={(e) => {
            args.onClose(e);
            setOpen(false);
          }}
        />
      </>
    );
  }
};
