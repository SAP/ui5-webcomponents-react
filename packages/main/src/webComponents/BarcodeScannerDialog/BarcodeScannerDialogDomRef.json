[
  { "name": "close", "visibility": "public", "description": "Closes the dialog and the scan session." },
  {
    "name": "show",
    "visibility": "public",
    "description": "Shows a dialog with the camera videostream. Starts a scan session."
  }
]
