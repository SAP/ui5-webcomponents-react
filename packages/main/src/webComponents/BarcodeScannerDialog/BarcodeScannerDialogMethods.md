## Methods

This component exposes public methods. You can invoke them directly on the instance of the component, e.g. by using React Refs.

| Name      | Parameters | Description                                                        |
| --------- | ---------- | ------------------------------------------------------------------ |
| **close** | &mdash;    | Closes the dialog and the scan session.                            |
| **show**  | &mdash;    | Shows a dialog with the camera videostream. Starts a scan session. |
