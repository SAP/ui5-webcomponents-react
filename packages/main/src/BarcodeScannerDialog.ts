import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { BarcodeScannerDialogDomRef, BarcodeScannerDialogPropTypes } from './webComponents/BarcodeScannerDialog';
import { BarcodeScannerDialog } from './webComponents/BarcodeScannerDialog';

deprecationNotice(
  'BarcodeScannerDialog',
  `importing from '@ui5/webcomponents-react/dist/BarcodeScannerDialog' is deprecated. Please use "import { BarcodeScannerDialog } from '@ui5/webcomponents-react';" instead.`
);

export { BarcodeScannerDialog };

export type { BarcodeScannerDialogPropTypes, BarcodeScannerDialogDomRef };
