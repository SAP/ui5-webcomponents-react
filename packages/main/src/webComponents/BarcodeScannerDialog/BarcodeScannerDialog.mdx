import { ControlsWithNote, DocsHeader, Footer } from '@sb/components';
import { Canvas, Meta } from '@storybook/blocks';
import * as ComponentStories from './BarcodeScannerDialog.stories';

<Meta of={ComponentStories} />

<DocsHeader since="0.18.0" />

<br />

## Example

<Canvas of={ComponentStories.Default} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<Footer />
