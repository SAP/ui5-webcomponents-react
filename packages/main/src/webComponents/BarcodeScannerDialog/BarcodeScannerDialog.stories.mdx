import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { BarcodeScannerDialog } from '@ui5/webcomponents-react/dist/BarcodeScannerDialog';
import { Button } from '@ui5/webcomponents-react/dist/Button';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { useRef } from 'react';

<Meta title="Inputs / BarcodeScannerDialog" component={BarcodeScannerDialog} />

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      const ref = useRef();
      return (
        <>
          <Button
            onClick={() => {
              ref.current.show();
            }}
          >
            Open BarcodeScannerDialog
          </Button>
          <BarcodeScannerDialog {...args} ref={ref} />
        </>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

## Methods

This component exposes public methods. You can invoke them directly on the instance of the component, e.g. by using React Refs.

| Name      | Parameters | Description                                                        |
| --------- | ---------- | ------------------------------------------------------------------ |
| **close** | &mdash;    | Closes the dialog and the scan session.                            |
| **show**  | &mdash;    | Shows a dialog with the camera videostream. Starts a scan session. |
