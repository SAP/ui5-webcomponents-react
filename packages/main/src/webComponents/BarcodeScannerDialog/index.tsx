import '@ui5/webcomponents-fiori/dist/BarcodeScannerDialog.js';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5CustomEvent } from '../../interfaces/Ui5CustomEvent';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface BarcodeScannerDialogAttributes {}

export interface BarcodeScannerDialogDomRef extends BarcodeScannerDialogAttributes, Ui5DomRef {
  /**
   * Closes the dialog and the scan session.
   */
  close: () => void;
  /**
   * Shows a dialog with the camera videostream. Starts a scan session.
   */
  show: () => void;
}

export interface BarcodeScannerDialogPropTypes extends BarcodeScannerDialogAttributes, CommonProps {
  /**
   * Fires when the scan fails with error.
   */
  onScanError?: (event: Ui5CustomEvent<HTMLElement, { message: string }>) => void;
  /**
   * Fires when the scan is completed successfuuly.
   */
  onScanSuccess?: (event: Ui5CustomEvent<HTMLElement, { text: string; rawBytes: Record<string, unknown> }>) => void;
}

/**
 * The `BarcodeScannerDialog` component provides barcode scanning functionality for all devices that support the `MediaDevices.getUserMedia()` native API. Opening the dialog launches the device camera and scans for known barcode formats.
 *
 * A `scanSuccess` event fires whenever a barcode is identified and a `scanError` event fires when the scan failed (for example, due to missing permisions).
 *
 * Internally, the component uses the zxing-js/library third party OSS. For a list of supported barcode formats, see the <ui5-link target="_blank" href="https://github.com/zxing-js/library">zxing-js/library</ui5-link> documentation
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/BarcodeScannerDialog" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const BarcodeScannerDialog = withWebComponent<BarcodeScannerDialogPropTypes, BarcodeScannerDialogDomRef>(
  'ui5-barcode-scanner-dialog',
  [],
  [],
  [],
  ['scan-error', 'scan-success']
);

BarcodeScannerDialog.displayName = 'BarcodeScannerDialog';

export { BarcodeScannerDialog };
