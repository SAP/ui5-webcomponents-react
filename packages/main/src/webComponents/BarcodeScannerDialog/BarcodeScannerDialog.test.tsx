import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { BarcodeScannerDialog } from '@ui5/webcomponents-react/dist/BarcodeScannerDialog';
import React from 'react';

describe('BarcodeScannerDialog', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<BarcodeScannerDialog />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(BarcodeScannerDialog);
});
