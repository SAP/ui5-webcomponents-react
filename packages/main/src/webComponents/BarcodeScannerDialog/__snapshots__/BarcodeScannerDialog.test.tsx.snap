// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarcodeScannerDialog Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-barcode-scanner-dialog
    ui5-barcode-scanner-dialog=""
  />
</DocumentFragment>
`;
